compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -937,7 +937,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             .get(ProvidedIdx::from_usize(arg_idx.index() - 1)
                         ) {
                             // Include previous comma
-                            span = prev.shrink_to_hi().to(span);
+                            // NOTE(#112437): we *manually* calculate the previous comma span here
+                            // because `Span::to` returns incomplete span when `prev` or `span`
+                            // involves macro contexts.
+                            span = Span::new(prev.hi(), span.hi(), span.ctxt(), span.parent());
```

Comment:
This fix looks incorrect if the argument comes from a macro variable, for example for the following the span will now start after 1 and go all the way to 2:
```rust
fn foo(_: i32) {}

macro_rules! m {
    ($e:expr) => {
        foo(1, $e)
    }
}

fn main() {
    m!(2)
}
```

I think you can swap the arguments to get a better span: `span.to(prev.shrink_to_hi())`.

Reply:
Unfortunately that isn't correct either: `Span::to` will also return incomplete span when the "current" argument span (`span`) is from a macro context...

e.g.

```rust
f(1, panic!());
```